docs/jwst/user_documentation/running_pipeline_python.rst

```diff
@@ -529,3 +529,196 @@ individual step parameter must be set when using this method, or else the coded
+
+
+::
+
+    # SampleScript2
+
+    import os
+    import traceback
+    import configparser
+    import multiprocessing
+    from glob import glob
+
+    def mk_stpipe_log_cfg(output_dir, log_name):
+        """
+        Create a configuration file with the name log_name, where
+        the pipeline will write all output.
+        Args:
+            outpur_dir: str, path of the output directory
+            log_name: str, name of the log to record screen output
+        Returns:
+            nothing
+        """
+        config = configparser.ConfigParser()
+        config.add_section("*")
+        config.set("*", "handler", "file:" + log_name)
+        config.set("*", "level", "INFO")
+        pipe_log_config = os.path.join(output_dir, "pipeline-log.cfg")
+        config.write(open(pipe_log_config, "w"))
+
+    def run_det1(uncal_file, output_dir):
+        """
+        Run the Detector1 pipeline on the given file.
+        Args:
+            uncal_file: str, name of uncalibrated file to run
+            outpur_dir: str, path of the output directory
+        Returns:
+            nothing
+        """
+        log_name = os.path.basename(uncal_file).replace('.fits', '')
+        mk_stpipe_log_cfg(output_dir, log_name+'.log')
+        from jwst.pipeline.calwebb_detector1 import Detector1Pipeline
+        pipe_success = False
+        try:
+            det1 = Detector1Pipeline()
+            det1.call(uncal_file, output_dir=output_dir, logcfg="pipeline-log.cfg", save_results=True)
+            pipe_success = True
+            print('\n * Pipeline finished for file: ', uncal_file, ' \n')
+        except Exception:
+            print('\n *** OH NO! The detector1 pipeline crashed! *** \n')
+            pipe_crash_msg = traceback.print_exc()
+        if not pipe_success:
+            crashfile = open(log_name+'_pipecrash.txt', 'w')
+            print('Printing file with full traceback')
+            print(pipe_crash_msg, file=crashfile)
+
+    def main():
+        input_data_dir = '/my_project_dir'
+        output_dir = input_data_dir
+
+        # get the files to run
+        files_to_run = glob(os.path.join(input_data_dir, '*_uncal.fits'))
+        print('Will run the pipeline on {} files'.format(len(files_to_run)))
+
+        # the output list should be the same length as the files to run
+        outptd = [output_dir for _ in range(len(files_to_run))]
+
+        # get the cores to use
+        cores2use = int(os.cpu_count()/2)   # half of all available cores
+        print('* Using ', cores2use, ' cores for multiprocessing.')
+
+        # set the pool and run multiprocess
+        p = multiprocessing.Pool(cores2use)
+        p.starmap(run_det1, zip(files_to_run, outptd))
+        p.close()
+        p.join()
+
+    if __name__ == '__main__':
+        multiprocessing.freeze_support()
+        main()
+        print('\n * Finished multiprocessing! \n')
+
+
+3. Using both, calling the pipeline with multiprocessing and turning on a step
```

Comment:
I don't think this should be listed as a third option.  IMO this information would be more appropriate inside of a warning block, with some wording like "This setup is not recommended, and may cause failures"

Reply:
Thanks, this is a good suggestion. I added the warning box.